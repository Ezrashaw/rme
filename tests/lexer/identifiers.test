2let2 
hello2world
hello_world

hello_let
hellolet

let fn true false