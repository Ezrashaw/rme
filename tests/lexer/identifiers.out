Sp(Literal(2.0), Span(0..1))
Sp(Keyword(Let), Span(1..4))
Sp(Literal(2.0), Span(4..5))
Sp(Identifier("hello"), Span(7..12))
Sp(Literal(2.0), Span(12..13))
Sp(Identifier("world"), Span(13..18))
Sp(Identifier("hello"), Span(19..24))
[1;3;31merror[0m: invalid start of token `_`

    hello_world
         [1;96m^[0m
Sp(Identifier("world"), Span(25..30))
Sp(Identifier("hello"), Span(32..37))
[1;3;31merror[0m: invalid start of token `_`

    hello_let
         [1;96m^[0m
Sp(Keyword(Let), Span(38..41))
Sp(Identifier("hellolet"), Span(42..50))
Sp(Keyword(Let), Span(52..55))
Sp(Keyword(Fn), Span(56..58))
Sp(Keyword(True), Span(59..63))
Sp(Keyword(False), Span(64..69))
